### DISCUSSION TITLE
[ROS2] How to subscribe image topic and visualize image data on nicegui?

### DISCUSSION THREAD

**Discussion by manhha1402** [zauberzeug/nicegui#3738] [Q&A] [answered]
### Description

Has anyone tried to subscribe image topic from ROS2 and visualize the image stream on nicegui?
Thank you very much

**Comment by python-and-novella** [comment]
I think you should create a discussion instead of issue.

**Comment by rodja** [comment]
Have you tried https://github.com/zauberzeug/nicegui/wiki/ROS2-image-display?

**Comment by manhha1402** [comment]
Hi @rodja, thank you for your example. I just tried it and got the error:
```
 from nicegui import app, globals, run, ui
ImportError: cannot import name 'globals' from 'nicegui' 
```
Could you please check and update the example?
Thank you

**Reply by rodja** [reply]
Sure. @JensOgorek will have a look at that.

**Reply by JensOgorek** [reply]
I will update the examples as a standalone Node tomorrow.

**Reply by JensOgorek** [reply, marked as answer]
I got it all to work on Ubuntu 24.04. @manhha1402 you are using the code copied from the wiki, correct? That code is outdated, since we don't use the globals import anymore.
Try the code from the [repository instead](https://github.com/zauberzeug/ros_nicegui_image_example). 
There is a small problem, with the reload dirs. When you start it via ros2 run, the uvicorn reload dirs will get ignored, and it tries to watch all of your system - same with the exclude filters. That is a problem that also happens within fastAPI and is most likely from coming from watchfiles itself. But unlike ros1, you can start the ros2 node directly with `python3 receiver.py` and everything will work. 
I will update examples to work without ros2 run and also the wiki with it.

**Reply by JensOgorek** [reply]
it's updated now

**Comment by manhha1402** [comment]
Hi @rodja and @JensOgorek , 
Thank you for your example. I am working on a ROS2 project that uses NiceGUI as a service and action client to send signals to server. If you have time, can you create an example how to use a nicegui as a service or action client?
